ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetProtocolSpecs.java

```diff
@@ -676,6 +679,15 @@ static ProtocolSpecBuilder shandongDefinition(
                     stackSizeLimit,
                     londonFeeMarket,
                     CoinbaseFeePriceCalculator.eip1559()))
+        .contractCreationProcessorBuilder(
+            (gasCalculator, evm) ->
+                new ContractCreationProcessor(
+                    gasCalculator,
+                    evm,
+                    true,
+                    List.of(MaxCodeSizeRule.of(contractSizeLimit), CachedInvalidCodeRule.of()),
```

Comment:
No.  London rejects all 0xEF code for all of London and Paris.  The EOF rule relaxes it and accepts EOF formatted code that starts with 0xEF.  It could be a configuration of the rule, but I read it cleaner to make it two rules.